### PR TITLE
add gradient function

### PR DESCRIPTION
**Pull request by KsanaKozlova**

### CODE REVIEW

File: dpnp/dpnp_algo/dpnp_algo_mathematical.pyx [hunk 1]
```diff
@@ -294,6 +294,27 @@ cpdef dparray dpnp_fmod(dparray x1, dparray x2):
     return call_fptr_2in_1out(DPNP_FN_FMOD, x1, x2, x1.shape)
 
 
+cpdef dparray dpnp_gradient(dparray y1, int dx=1):
+
+    len = y1.size
```
Comment (densmirn) [inline dpnp/dpnp_algo/dpnp_algo_mathematical.pyx:299]: `len` is built-in name. Let's rename the variable, e.g. `size`.

File: dpnp/dpnp_iface_mathematical.py [hunk 2]
```diff
@@ -787,6 +788,46 @@ def fmod(x1, x2, **kwargs):
...
+    [1.0, 1.5, 2.5, 3.5, 4.5, 5.0]
+    >>> result = np.gradient(y, 2)
+    >>> [x for x in result]
+    [0.5, 0.75, 1.25, 1.75, 2.25, 2.5]
+
+    """
+    if not use_origin_backend(y1) and not kwargs:
+        if not isinstance(y1, dparray):
+            pass
+        elif len(varargs) != 0 and not isinstance(varargs[0], int):
```
Comment (densmirn) [inline dpnp/dpnp_iface_mathematical.py:820]: What if `len(varargs) = 2 and varargs[0] = 2`? Don't we need to fallback to `numpy` in this case?

File: tests/test_mathematical.py [hunk 3]
```diff
@@ -143,3 +143,29 @@ def test_cross_3x3(self, x1, x2):
...
         expected = numpy.cross(x1_, x2_)
         numpy.testing.assert_array_equal(expected, result)
+
+
+class TestGradient:
+
+    @pytest.mark.parametrize("array", [[2, 3, 6, 8, 4, 9],
+                                       [3., 4., 7.5, 9.],
+                                       [2, 6, 8, 10]])
+    def test_gradient_y1(self, array):
```
Comment (densmirn) [inline tests/test_mathematical.py:153]: Both tests are named the same.

**Review by densmirn — APPROVED** [review]

File: dpnp/dpnp_algo/dpnp_algo_mathematical.pyx [hunk 4]
```diff
@@ -294,6 +294,27 @@ cpdef dparray dpnp_fmod(dparray x1, dparray x2):
...
+   
+    cur = (y1[1] - y1[0]) / dx
+
+    result._setitem_scalar(0, cur)
+
+    cur = (y1[-1] - y1[-2]) / dx
+
+    result._setitem_scalar(size - 1, cur)
+  
+    for i in range(1, len - 1):
```
Comment (densmirn) [inline dpnp/dpnp_algo/dpnp_algo_mathematical.pyx:311]:
```suggestion
    for i in range(1, size-1):
```

**Comment by shssf** [comment]
Could you please rebase on master?

**Review by shssf — APPROVED** [review]

**Comment by shssf** [comment]
error in tests:
```
NameError: name 'dpnp_gradient' is not defined
```

**Pull request merged**